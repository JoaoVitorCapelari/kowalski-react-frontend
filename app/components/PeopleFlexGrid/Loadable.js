/**
 *
 * Asynchronously loads the component for PeopleFlexGrid
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
