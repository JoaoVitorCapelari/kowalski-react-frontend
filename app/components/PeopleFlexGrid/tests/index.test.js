import React from 'react';
import { shallow, mount } from 'enzyme';
import { fromJS } from 'immutable';

import PeopleFlexGrid from '../index';

describe('<PeopleFlexGrid />', () => {
  it('renders', () => {
    const renderedComponent = shallow(
      <PeopleFlexGrid people={fromJS([{ kUserId: 1, name: 'Test' }])} />
    );
    expect(renderedComponent.html()).toMatchSnapshot();
  });

  it('callbacks personSelected properly', () => {
    const personSelect = jest.fn();
    const renderedComponent = mount(
      <PeopleFlexGrid
        personSelected={personSelect}
        people={fromJS([{ kUserId: 1, name: 'Test' }])}
      />
    );
    renderedComponent.find('div[role="button"]').simulate('click');
    expect(personSelect).toHaveBeenCalledWith(1);
  });
});
