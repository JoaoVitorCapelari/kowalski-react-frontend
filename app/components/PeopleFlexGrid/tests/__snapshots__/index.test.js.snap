// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeopleFlexGrid /> renders 1`] = `"<div class=\\"PeopleFlexGrid__Container-lDUyI bSnzxM\\"><div class=\\"PeopleFlexGrid__PersonContainer-jLvaEV kMyiMo\\" tabindex=\\"0\\" role=\\"button\\"><div class=\\"PeopleFlexGrid__PersonPictureContainer-bQjxpx lhHdjA\\"><img class=\\"PeopleFlexGrid__PersonImg-cbMrBq jgRTBA\\" src=\\"[object Object]\\"/></div><span class=\\"PeopleFlexGrid__PersonNameContainer-sZXHU iGYrwP\\">Test</span></div></div>"`;
